pennylane_qiskit/qiskit_device2.py

```diff
@@ -0,0 +1,606 @@
+            compiled_circ.name = f"circ{len(compiled_circuits)}"
+            compiled_circuits.append(compiled_circ)
+
+        return compiled_circuits
+
+    # pylint: disable=unused-argument
+    def execute(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ) -> Result_or_ResultBatch:
+        session = self._session or Session(backend=self.backend)
+
+        if not self._use_primitives:
+            results = self._execute_runtime_service(circuits, session=session)
+            return results
+
+        results = []
+
+        if isinstance(circuits, (QuantumTape, QuantumScript)):
+            circuits = [circuits]
+
+        for circ in circuits:
+            if isinstance(circ.measurements[0], (ExpectationMP, VarianceMP)):
+                execute_fn = self._execute_estimator
+            elif isinstance(circ.measurements[0], ProbabilityMP):
+                execute_fn = self._execute_sampler
+            else:
+                execute_fn = self._execute_runtime_service
+            results.append(execute_fn(circ, session))
+
+        if self._session is None:
+            # if this was not a session set on the device, but just one created for this execution, close
+            session.close()
+
+        return results
+
+    def _execute_runtime_service(self, circuits, session):
+        """Execution using old runtime_service (can't use runtime sessions)"""
+
+        # update kwargs in case Options has been modified since last execution
+        self._update_kwargs()
+
+        # in case a single circuit is passed
+        if isinstance(circuits, (QuantumTape, QuantumScript)):
+            circuits = [circuits]
+
+        qcirc = [
+            circuit_to_qiskit(circ, self.num_wires, diagonalize=True, measure=True)
+            for circ in circuits
+        ]
+        compiled_circuits = self.compile_circuits(qcirc)
+
+        program_inputs = {"circuits": compiled_circuits, "shots": self.shots.total_shots}
+
+        for kwarg in self._kwargs:
+            program_inputs[kwarg] = self._kwargs.get(kwarg)
+
+        options = {
+            "backend": self.backend.name,
+            "log_level": self.options.environment.log_level,
+            "job_tags": self.options.environment.job_tags,
+            "max_execution_time": self.options.max_execution_time,
+        }
+
+        # Send circuits to the cloud for execution by the circuit-runner program.
+        job = self.service.run(
+            program_id="circuit-runner",
+            options=options,
+            inputs=program_inputs,
+            session_id=session.session_id,
+        )
+        self._current_job = job.result(decoder=RunnerResult)
+
+        results = []
+
+        for index, circuit in enumerate(circuits):
+            self._samples = self.generate_samples(index)
+            res = [
+                mp.process_samples(self._samples, wire_order=self.wires)
```

Comment:
do we need to save it to the device?
```suggestion
            samples = self.generate_samples(index)
            res = [
                mp.process_samples(samples, wire_order=self.wires)
```